Display Event Creator Name in Home Screen Event Cards

rita-naimi opened the pull request:
## What?

Adds event creator name display to event cards on the home screen, showing "by [First name] [Last name]" or "[Organization name]" for each event.

## Why?

Event cards currently don't show who created the event, making it difficult for users to identify events from people they know or trust.

## How?

Added string resource and modified `EventCard` to fetch and display creator information asynchronously using `UserRepositoryProvider`.


Fixes #391

<img width="261" height="217" alt="Capture d’écran 2025-12-16 à 23 29 54" src="https://github.com/user-attachments/assets/333e72c8-d21e-42ed-90dc-ec386ab85939" />

rita-naimi committed:
feat(home): display event creator name in event cards

# Conflicts:
#	app/src/main/java/com/github/se/studentconnect/ui/utils/ListOfEvents.kt

TristanBour commented on app/src/main/java/com/github/se/studentconnect/ui/utils/ListOfEvents.kt:
```diff
@@ -248,6 +249,19 @@ fun EventCard(
+  // Fetch event creator information
+  val userRepository = UserRepositoryProvider.repository
+  val creator by
+      produceState<com.github.se.studentconnect.model.user.User?>(
```
You should import User and use just that instead of the full path

TristanBour reviewed (changes requested):
Great small addition, just a fast change to be done and it should be perfect.

rita-naimi committed:
feat(events): display organization name in event cards

- Show organization name instead of user name when event is created by an organization
- Replace all hardcoded dp values with constants from Dimensions object
- Add new dimension constants for event card components (image height, padding, chip styling, live badge)
- Refactor EventCard to fetch and display organization data when organizationId is present

rita-naimi committed:
fix(events): correct typo and add missing Dimensions import

- Fix typo: EvenevetCardCornerRadius -> EventCardCornerRadius in EventView.kt
- Add missing import for Dimensions object in ListOfEvents.kt

rita-naimi committed:
refactor(events): use imported classes instead of fully qualified names

- Import User and Organization classes
- Replace fully qualified class names with simple imports in produceState declarations

rita-naimi committed:
chore(format): apply ktfmt formatting

TristanBour reviewed (approved):
Perfect, nice job!

rita-naimi committed:
chore(format): apply ktfmt formatting
